src/main/scala/com/rbmhtechnology/calliope/serializer/SequencedEventSerializer.scala

```diff
@@ -61,10 +64,14 @@ abstract class SequencedEventSerializer(system: ExtendedActorSystem, payloadSeri
       .setCreationTimestamp(sequencedEvent.creationTimestamp.toEpochMilli)
 
   def sequencedEvent(format: SequencedEventFormat): SequencedEvent[_] =
-    SequencedEvent(
-      payloadSerializer.payload(format.getPayload),
-      format.getSourceId,
-      format.getSequenceNr,
-      Instant.ofEpochMilli(format.getCreationTimestamp)
-    )
+    try {
+      SequencedEvent(
+        payloadSerializer.payload(format.getPayload),
+        format.getSourceId,
+        format.getSequenceNr,
+        Instant.ofEpochMilli(format.getCreationTimestamp)
+      )
+    } catch {
+      case NonFatal(err) => throw new PayloadNotSerializableException("Unable to deserialize payload", err, format.getPayload)
```

Comment:
Keep it as simple as possible at the moment. We will later add generic support for protocol stacks where we need to rethink error handling and reporting. But if you're more comfortable with your current proposal/solution let's keep it as a temporary solution.